Jenkinsfile

```diff
@@ -1,136 +1,148 @@
-                                         description: 'All unit tests passed',
-                                         targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                } catch(Exception ex) {
-                    if (env.CHANGE_ID) {
-                        pullRequest.createStatus(status: 'failure',
-                                     context: 'jenkins/unittests',
-                                     description: 'unit tests failed',
-                                     targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                    currentBuild.result = 'UNSTABLE'
-                }
-            }
+      steps {
+        script {
+          try {
+            sh './docker/env/hydra.sh unit-tests'
+            markPullRequest('success', 'jenkins/unittests', 'All unit tests are passed')
+          } catch(Exception ex) {
+            markPullRequest('failure', 'jenkins/unittests', 'Some unit tests failed')
+          }
         }
+      }
     }
     stage("lint test-cases") {
-        steps {
-            script {
-                try {
-                    sh ''' ./utils/lint_test_cases.sh '''
-
-                    if (env.CHANGE_ID) {
-                        pullRequest.createStatus(status: 'success',
-                                         context: 'jenkins/lint_test_cases',
-                                         description: 'all test cases are checked',
-                                         targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                } catch(Exception ex) {
-                    if (env.CHANGE_ID) {
-                        pullRequest.createStatus(status: 'failure',
-                                         context: 'jenkins/lint_test_cases',
-                                         description: 'some test cases failed to check',
-                                         targetUrl: "${env.JOB_URL}/workflow-stage")
-                    }
-                    currentBuild.result = 'UNSTABLE'
-                }
-            }
+      steps {
+        script {
+          try {
+            sh ''' ./utils/lint_test_cases.sh '''
+            markPullRequest('success', 'jenkins/lint_test_cases', 'All test cases are passed')
+          } catch(Exception ex) {
+            markPullRequest('failure', 'jenkins/lint_test_cases', 'Some test cases failed')
+          }
         }
+      }
     }
     stage("provision test") {
-        when { changeRequest() }
-        steps {
+      when {
+        expression {
+          return lookForLabels("test-provision,test-provision-aws,test-provision-gce") && currentBuild.result == null
+        }
+      }
+      failFast true
+      parallel {
+        stage('provision test on gce') {
+          when {
+            expression {
+              return lookForLabels("test-provision,test-provision-gce")
+            }
+          }
+          steps {
             node('aws-sct-builders-eu-west-1') {
```

Comment:
Use gce builders. So it won't be cross cloud setup. We don't want to pay that extra on the traffic

Reply:
Fixed